gson/src/test/java/com/google/gson/internal/bind/DefaultDateTypeAdapterTest.java

```diff
@@ -223,6 +225,27 @@ public void testUnexpectedToken() throws Exception {
     }
   }
 
+  @Test
+  public void testGsonDateFormat() {
+    // Set the default timezone to UTC
+    TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+    Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd HH:mm z").create();
+    Date originalDate = new Date(0);
+
+    // Serialize the date object
+    String json = gson.toJson(originalDate);
+    assertEquals("\"1970-01-01 00:00 UTC\"", json);
+
+    // Deserialize a date string with the PST timezone
+    Date deserializedDate = gson.fromJson("\"1970-01-01 00:00 PST\"", Date.class);
```

Comment:
Maybe add an assertion here that the date was parsed correctly, I assume it would be this:
```suggestion
    Date deserializedDate = gson.fromJson("\"1970-01-01 00:00 PST\"", Date.class);
    assertThat(deserializedDate).isEqualTo(new Date(28800000));
```